### DISCUSSION TITLE
Looking for annotation similar to "nginx.ingress.kubernetes.io/auth-url"

### DISCUSSION THREAD

**Discussion by chatgptfighter** [nginx/kubernetes-ingress#5534] [Q&A] [answered]
Hi, we migrating project from kubernetes/ingress-nginx/ to nginxinc/kubernetes-ingress. Stuck a bit with annotations:
"nginx.ingress.kubernetes.io/auth-url"
"nginx.ingress.kubernetes.io/auth-response-headers"
Somehow cannot find analogs here. Need community help

**Comment by jasonwilliams14** [comment]
@chatgptfighter Hello there.
We are currently working on an implementation to use `auth_request` with our project.
you can see we started an issue to discuss:

https://github.com/nginxinc/kubernetes-ingress/issues/4667

**Reply by pdabelf5** [reply]
@chatgptfighter until https://github.com/nginxinc/kubernetes-ingress/issues/4667 is developed and released, snippets may be a good option for you
https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-snippets/
https://nginx.org/en/docs/http/ngx_http_auth_request_module.html

**Reply by chatgptfighter** [reply, marked as answer]
Thx all, Below you can find solution that applied in our end and it probably will save someone's time:
for api clients ingress (service requires authorisation):
```
nginx.org/location-snippets: |
      auth_request "/someauth"
      auth_request_set $authHeader0 $upstream_header;
      proxy_set_header 'x-some-header' $authHeader0;
```
for api server ingress (config that provide authorisation):
```
nginx.org/server-snippets: |
      location = /someauth{
        internal;
        proxy_set_header X-Auth-Request-Redirect $request_uri;
        proxy_set_header X-Original-Method $request_method;
        proxy_pass https://someserver.com/authpath;
      }
```
